src/TelemetryChannels/ServerTelemetryChannel/Shared/Implementation/TransmissionSender.cs

```diff
@@ -100,15 +164,105 @@ protected void OnTransmissionSent(TransmissionProcessedEventArgs args)
                 int currentCapacity = Interlocked.Decrement(ref this.transmissionCount);
                 if (exception == null)
                 {
-                    TelemetryChannelEventSource.Log.TransmissionSentSuccessfully(transmission.Id, currentCapacity);
+                    TelemetryChannelEventSource.Log.TransmissionSentSuccessfully(acceptedTransmission.Id, currentCapacity);
+
+                    if (responseContent == null && exception is WebException)
+                    {
+                        var response = (HttpWebResponse)((WebException)exception).Response;
+                        responseContent = new HttpWebResponseWrapper()
+                        {
+                            StatusCode = (int)response.StatusCode,
+                            StatusDescription = response.StatusDescription,
+                            RetryAfterHeader = response.Headers.Get("Retry-After")
+                        };
+                    }
                 }
                 else
                 {
-                    TelemetryChannelEventSource.Log.TransmissionSendingFailedWarning(transmission.Id, exception.ToString());
+                    TelemetryChannelEventSource.Log.TransmissionSendingFailedWarning(acceptedTransmission.Id, exception.ToString());
+                }
+
+                this.OnTransmissionSent(new TransmissionProcessedEventArgs(acceptedTransmission, exception, responseContent));
+            }
+        }
+
+        /// <summary>
+        /// Checks if the transmission throttling policy allows for sending another request.
+        /// If so, this method will add a request to the current throttle count (unless peeking)
+        /// </summary>
+        /// <returns>The number of events that are sendable</returns>
+        private int IsTransmissionSendable(int numEvents = 1, bool peek = false)
+        {
+            if (!this.applyThrottle)
+            {
+                return numEvents;
+            }
+
+            var throttleWindowId = (long)(DateTime.UtcNow.Subtract(new DateTime(1970, 1, 1)).TotalMilliseconds / this.ThrottleWindow);
+            var isInThrottleWindow = throttleWindowId == currentThrottleWindowId;
+            if (isInThrottleWindow && currentItemsCount < this.ThrottleLimit)
+            {
+                var numAccepted = Math.Min(numEvents, this.ThrottleLimit - this.currentItemsCount);
```

Comment:
Can this.ThrottleLimit - this.currentItemsCount never be negative (since this appears to be inside a check that currentItemsCount is less than the throttle limit)?